src/java/org/apache/cassandra/db/filter/RowFilter.java

```diff
@@ -604,7 +609,7 @@ public static abstract class Expression
         // and this is why we have some UNUSEDX for values we don't use anymore
         // (we could clean those on a major protocol update, but it's not worth
         // the trouble for now)
-        protected enum Kind { SIMPLE, MAP_EQUALITY, UNUSED1, CUSTOM, USER }
+        protected enum Kind { SIMPLE, MAP_EQUALITY, UNUSED1, CUSTOM, USER, VECTOR_RADIUS }
```

Comment:
In terms of making divergent change from upstream that may cause problems later, this one feels like the most brittle. As the comment indicates, the enum order is used for serde. If there are future conflicts where upstream adds an enum, we could run into issues.

One hack could be to misuse the `MAP_EQUALITY` kind. In that case, we would encode it as `column[:querypoint] < :distance`. Then, we'd essentially merge the `GeoDistanceExpression` into the `MapEqualityExpression`. (Note that I am expanding the `MapEqualityExpression` with #804 so that it supports range expressions.)

I don't think we need that hack, but it seems like it is worth calling out.

Reply:
I added a new ordinal, like you suggested.